lib/ts/recipe/anomalydetection/types.ts

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import OverrideableBuilder from "supertokens-js-override";
+
+export type TypeInput = {
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type TypeNormalisedInput = {
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type APIInterface = {};
+
+export type RecipeInterface = {
+    checkForAnomaly: (input: {
+        request: any;
+        userId: string;
+        userContext: any;
+    }) => Promise<{
+        status: "ANOMALY_DETECTED" | "NO_ANOMALY_DETECTED" | "ANOMALY_SERVICE_ERROR";
+        anomalyDetected: boolean;
+        message: string;
```

Comment:
why do we need `anomalyDetected: boolean;` when we have the status of `ANOMALY_DETECTED`? or vice  versa?